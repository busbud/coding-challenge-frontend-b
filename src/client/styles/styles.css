* {
  box-sizing: border-box;
}

html,
body,
#root,
.main-container {
  height: 100%;
}

body {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.card {
  display: flex;
  flex-flow: column;
  justify-content: space-between;
  margin: 10px 50px;
  padding: 15px 30px;
  width: 70%;
  max-width: 850px;
  min-height: 180px;
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(255, 255, 255);
  border-radius: 15px;
  box-shadow: 2px 2px 3px #c9c9c9;
}

.card p {
  margin: 7px 0;
  padding: 0;
  font-size: 18px;
}

.card-middle {
  display: flex;
  flex-flow: column nowrap;
  justify-content: flex-start;
}

div.card-departure p,
div.card-arrival p {
  display: inline-block;
}

.card-departure {
  color: rgb(56, 55, 66);
  font-weight: 700;
}

.card-arrival {
  color: rgb(85, 83, 99);
}

.card-top {
  display: flex;
  justify-content: space-between;
}

.card-operator-logo {
  max-width: 100px;
}

p.card-price {
  display: inline-block;
  color: #127ccb;
  font-size: 25px;
  font-weight: bold;
  top: 10px;
  right: 10px;
}

p.card-duration {
  color: rgb(117, 117, 117);
}

.fa-icon {
  margin-right: 5px;
}

.card#loading-card {
  padding-top: 25px;
}

.search-input-form {
  display: flex;
  flex-flow: row wrap;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 20px;
}

button:focus {
  outline: none;
}

input:hover {
  cursor: pointer;
  text-decoration: none;
}

.search-button {
  width: 200px;
  height: 50px;
  line-height: 50px;
  background-color: #ff5c60;
  border-style: none;
  color: white;
  font-family: inherit;
  font-size: inherit;
}

.search-button:hover {
  background-color: #d14c4e;
  cursor: pointer;
  text-decoration: none;
}

.form-input {
  width: 200px;
  height: 50px;
  line-height: 50px;
  background-color: white;
  text-align: center;
  border-right: 1px solid rgb(180, 180, 180);
  cursor: pointer;
}

#date-input {
  padding-left: 20px;
  border-style: none;
  font-family: inherit;
  font-size: inherit;
}

.search-input-form > *:first-child {
  border-radius: 10px 0 0 10px;
}

.search-input-form > *:last-child {
  border-radius: 0 10px 10px 0;
}

.main-container {
  display: flex;
  flex-flow: column;
}

.result-container {
  padding: 20px 0;
  background-color: #7cc9d0;
  flex-grow: 1;
  display: flex;
  flex-flow: column;
  align-items: center;
}

#osheaga-schedule {
  max-width: 1000px;
  width: 80%;
  box-sizing: inherit;
  border: 10px solid white;
}

.search-container {
  display: flex;
  background-color: rgb(176, 201, 255);
  padding: 40px;
  flex-flow: column;
  align-items: center;
  background-image: url(https://www.osheaga.com/uploads/osheaga/backgrounds/Osheaga-homecoming.jpb.jpg?v=67a753a108ff7d65f3d9355c44518fdf);
  background-position-x: center;
  background-position-y: 45%;
}

.logo-image {
  margin: 30px;
  max-width: 400px;
  width: 50%;
  display: block;
}

@media screen and (max-width: 480px) {
  .search-button,
  .form-input,
  #date-input {
    margin: 0.1em;
    font-size: 10px;
    width: 100%;
    height: 30px;
    line-height: 30px;
    display: block;
    border-radius: 1em !important;
  }

  .searh-input-form {
    display: flex;
    flex-flow: column;
  }

  p.card-price,
  .card p {
    font-size: 10px;
  }

  .card-operator-logo {
    max-width: 70px;
  }

  .card {
    overflow-x: scroll;
  }
}
