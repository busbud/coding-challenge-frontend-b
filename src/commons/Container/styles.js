import styled from "styled-components";

import { BREAKPOINTS } from "utils/constants";

export const StyledContainer = styled.div`
  width: 100%;
  padding: 0 16px;

  @media screen and (min-width: ${BREAKPOINTS.tablet}) {
    width: 980px;
    padding: 0;
    margin: 0 auto;
  }
`;
