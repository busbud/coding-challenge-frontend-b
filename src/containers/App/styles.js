export default theme => ({
  root: {
    flexGrow: '1',
    backgroundColor: theme.palette.background.default,
  },
});
