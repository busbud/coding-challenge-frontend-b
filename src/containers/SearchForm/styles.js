import red from 'material-ui/colors/red';

export default {
  textField: {
    width: '100%',
  },
  button_wrapper: {
    position: 'relative',
  },
  button: {
    position: 'relative',
    zIndex: 1,
  },
  circularProgress: {
    position: 'absolute',
    top: -8,
    left: -8,
    color: red[500],
  },
};
