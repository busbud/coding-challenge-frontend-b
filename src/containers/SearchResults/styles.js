export default {
  root: {
    marginTop: '6rem',
    marginBottom: '3rem',
  },
};
