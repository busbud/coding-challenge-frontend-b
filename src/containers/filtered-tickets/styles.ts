// Packages
import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spaces.tiny};
    section {
      width: min(60rem, 100%);
    }

    ${media.greaterThan('medium')`
      section {
        width: min(110rem, 100%);
      }
    `}
  `}
`

export const Image = styled.img`
  width: min(15rem, 100%);

  ${media.greaterThan('medium')`
    margin: 0;
    width: min(18rem, 100%);
  `}
`

export const BusInfo = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 2fr 1fr;
    text-align: center;
    column-gap: ${theme.spaces.small};
    row-gap: ${theme.spaces.small};
    font-size: calc(${theme.spaces.xsmall} - 2px);

    .svg__arrow {
      font-size: 3rem;
      margin: 0 2rem 0 2rem;
    }

    .svg__circle {
      margin: 0 16px;
      width: 0.5rem;
    }
  `}
`

export const Text = styled.span<{ bold?: boolean; margin?: string }>`
  ${({ theme, bold, margin }) => css`
    font-size: ${theme.font.sizes.small};
    line-height: ${theme.font.sizes.small};
    font-weight: ${bold && 500};
    margin: ${margin};

    &.text__medium {
      font-size: ${theme.font.sizes.xmedium};
    }
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xmedium};
    line-height: ${theme.font.sizes.medium};
    font-weight: 500;
    text-align: center;
    margin-bottom: ${theme.spaces.large};
    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `};
  `}
`

export const ContentEmptyState = styled(Wrapper)`
  ${media.greaterThan('medium')`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  `};
`

export const EmptyState = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr;
    column-gap: ${theme.spaces.small};
    row-gap: ${theme.spaces.small};
    font-size: calc(${theme.spaces.xsmall} - 2px);

    ${media.greaterThan('medium')`
       grid-template-columns: 1fr 1fr;
    `};
  `}
`
