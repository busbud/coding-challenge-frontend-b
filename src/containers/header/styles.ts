import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Logo = styled.img`
  width: 14.5rem;

  ${media.greaterThan('medium')`
    width: 12rem;
  `}
`

export const Wrapper = styled.header`
  ${({ theme }) => css`
    text-align: center;
    box-sizing: border-box;
    box-shadow: ${theme.shadow.small};
    padding: ${theme.spaces.small};

    ${media.greaterThan('medium')`
      text-align: left;
      margin: auto;
    `}
  `}
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  margin: 4.6rem auto 0;

  ${media.greaterThan('medium')`
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    margin: 15rem auto;
  `}
`
