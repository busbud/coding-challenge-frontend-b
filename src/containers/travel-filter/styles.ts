// Packages
import styled, { css } from 'styled-components'
import { rgba } from 'polished'
import media from 'styled-media-query'

// Common Layout
import { Box as LBox } from 'layout/box'

export const Box = styled(LBox)`
  ${({ theme }) => css`
    flex-direction: column;
    > div {
      flex: 1 1 20px;
      margin: 0.5rem;
      width: 100%;
    }

    > button {
      height: 36px;
      width: 36px;
      margin: -18px;
      padding: 5px;
      transition: ${theme.transition.ease};
      z-index: 1;
      background: ${theme.colors.white};
      border: 1px solid ${rgba(theme.colors.muted, 1)};
      color: ${theme.colors.gray};
      font-size: ${theme.font.sizes.xmedium};

      :hover {
        border: 2px solid ${theme.colors.primary};
      }
    }

    .DayPicker {
      width: 100%;
    }

    ${media.greaterThan('medium')`
      flex-direction: row;
      > div {
        flex: 1 1 100px;
      }

      > button {
        width: 46px;
        height: 46px;
        margin: 11px -13px;
      }
    `}
  `}
`

export const Row = styled(LBox)`
  flex-direction: column;
  max-width: 100%;
  ${media.greaterThan('medium')`
    flex-direction: row;
    > div {
      max-width: 32%;
    }
  `}
`

export const FloatButton = styled.div`
  ${({ theme }) => css`
    left: 0;
    position: absolute;
    text-align: center;
    width: 100%;
    margin-top: 1.2rem;

    > button {
      font-size: ${theme.font.sizes.small};
      box-shadow: ${theme.shadow.medium};
    }

    svg {
      margin-right: ${theme.spaces.xxsmall};
    }
  `}
`
