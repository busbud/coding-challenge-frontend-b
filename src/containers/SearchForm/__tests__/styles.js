import styles from '../styles';

describe('component | SearchForm | styles', () => {
  it('should be defined', () => {
    // then
    expect(styles).toBeDefined();
  });

  it('should match snapshot', () => {
    // then
    expect(styles).toMatchSnapshot();
  });
});
