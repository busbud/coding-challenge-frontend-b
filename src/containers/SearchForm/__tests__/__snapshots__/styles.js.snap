// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component | SearchForm | styles should match snapshot 1`] = `
Object {
  "button": Object {
    "position": "relative",
    "zIndex": 1,
  },
  "button_wrapper": Object {
    "position": "relative",
  },
  "circularProgress": Object {
    "color": "#f44336",
    "left": -8,
    "position": "absolute",
    "top": -8,
  },
  "textField": Object {
    "width": "100%",
  },
}
`;
